pkg/ruler/scheduler.go

```diff
@@ -206,7 +209,16 @@ func (s *scheduler) addNewConfigs(now time.Time, cfgs map[string]configs.Version
 	hasher := fnv.New64a()
 
 	for userID, config := range cfgs {
-		rulesByFilename, err := config.Config.Parse()
+		rulesByGroup := map[string][]rules.Rule{}
+		var err error
+		switch s.ruleFormatVersion {
+		case configs.RuleFormatV1:
+			rulesByGroup, err = config.Config.ParseV1()
+		case configs.RuleFormatV2:
+			rulesByGroup, err = config.Config.ParseV2()
+		default:
+			panic("unknown rule format")
```

Comment:
hmm

Reply:
Panics on programming errors are pretty normal/ok, no? Best way to actually catch them quickly...